packages/@aws-cdk/aws-events/lib/base-event-bus.ts

```diff
@@ -0,0 +1,130 @@
+   */
+  readonly eventBusArn: string;
+
+  /**
+   * The JSON policy of this event bus resource
+   *
+   * @attribute
+   * @link https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-events-eventbus.html#Policy-fn::getatt
+   */
+  readonly eventBusPolicy: string;
+
+  /**
+   * The partner event source to associate with this event bus resource
+   *
+   * @attribute
+   * @link https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-events-eventbus.html#cfn-events-eventbus-eventsourcename
+   */
+  readonly eventSourceName?: string;
+}
+
+export interface BaseEventBusProps {
+  /**
+   * The name of the event bus you are creating.
+   */
+  readonly eventBusName: string;
+
+  /**
+   * The name of the partner event source.
+   * @default - No partner event source.
+   */
+  readonly eventSourceName?: string;
+}
+
+/**
+ * Define a CloudWatch EventBus
+ *
+ * @resource AWS::Events::EventBus
+ */
+export abstract class BaseEventBus extends Resource implements IEventBus {
+
+  /**
+   * Import an existing event bus resource
+   * @param scope Parent construct
+   * @param id Construct ID
+   * @param eventBusArn ARN of imported event bus
+   */
+  public static fromEventBusArn(scope: Construct, id: string, eventBusArn: string): IEventBus {
+    const parts = Stack.of(scope).parseArn(eventBusArn);
+
+    class Import extends Resource implements IEventBus {
+      public readonly eventBusArn = eventBusArn;
+      public readonly eventBusName = parts.resourceName || '';
+      public readonly eventBusPolicy = '';
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * The physical ID of this event bus resource
+   */
+  public readonly eventBusName: string;
+
+  /**
+   * The ARN of the event bus, such as:
+   * arn:aws:events:us-east-2:123456789012:event-bus/aws.partner/PartnerName/acct1/repo1.
+   */
+  public readonly eventBusArn: string;
+
+  /**
+   * The policy for the event bus in JSON form.
+   */
+  public readonly eventBusPolicy: string;
+
+  /**
+   * The name of the partner event source
+   */
+  public readonly eventSourceName?: string;
+
+  constructor(scope: Construct, id: string, props: BaseEventBusProps) {
+    super(scope, id, {
+      physicalName: props.eventBusName
+    });
+
+    const { eventBusName, eventSourceName } = props;
+    this.validateName(eventBusName);
+
+    const eventBus = new CfnEventBus(this, 'Resource', {
+      name: eventBusName,
+      eventSourceName
+    });
+
+    this.eventBusArn = this.getResourceArnAttribute(eventBus.attrArn, {
+      service: 'events',
+      resource: 'event-bus',
+      resourceName: eventBus.name
+    });
+
+    this.eventBusName = this.getResourceNameAttribute(eventBus.ref);
+    this.eventBusPolicy = eventBus.attrPolicy;
+    this.eventSourceName = eventBus.eventSourceName;
+  }
+
+  private validateName(eventBusName: string) {
```

Comment:
I'd rather use `validate()` here, but since `this.eventBusName` is assigned using `this.getResourceNameAttribute()` I don't think I can do the same regex check (without assigning the passed in value to some internal variable). Is there some way that this is commonly done in code?

Reply:
No, but validating user input the way you're doing and throwing an error seems perfectly sane to me.